### PR TITLE
Remove testmode from ApplicationEngine

### PR DESCRIPTION
**Pull request by shargon**

### CODE REVIEW

File: src/neo/SmartContract/ApplicationEngine.cs [hunk 1]
```diff
@@ -112,8 +111,16 @@ protected override void ContextUnloaded(ExecutionContext context)
         }
 
         public static ApplicationEngine Create(TriggerType trigger, IVerifiable container, StoreView snapshot, long gas, bool testMode = false)
```
Comment (erikzhang) [inline src/neo/SmartContract/ApplicationEngine.cs:113]: The parameters of the `Create()` method must be consistent with the constructor.

Comment (shargon) [reply src/neo/SmartContract/ApplicationEngine.cs]: Done, please take a look

File: src/neo/SmartContract/ApplicationEngine.cs [hunk 2]
```diff
@@ -111,9 +110,17 @@ protected override void ContextUnloaded(ExecutionContext context)
             }
         }
 
-        public static ApplicationEngine Create(TriggerType trigger, IVerifiable container, StoreView snapshot, long gas, bool testMode = false)
-            => applicationEngineProvider?.Create(trigger, container, snapshot, gas, testMode)
-                ?? new ApplicationEngine(trigger, container, snapshot, gas, testMode);
+        public static long TestGas(StoreView snapshot = null)
```
Comment (erikzhang) [inline src/neo/SmartContract/ApplicationEngine.cs:113]: Just add a big enough constant? We don't need a function.

Comment (shargon) [reply src/neo/SmartContract/ApplicationEngine.cs:113]: Without the snapshot? 9000 of Gas?

File: src/neo/SmartContract/ApplicationEngine.cs [hunk 3]
```diff
@@ -25,13 +26,17 @@ private class InvocationState
             public bool NeedCheckReturnValue;
         }
 
+        /// <summary>
+        /// This constant can be used for testing scripts, by default has the same value as max block system fee.
+        /// </summary>
+        public const long TestGas = 9000_00000000;
```
Comment (erikzhang) [inline src/neo/SmartContract/ApplicationEngine.cs:32]: `20`?

**Pull request merged**